From 23ad893d02681853cf3b51096940f7e1364113af Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Wed, 25 Apr 2018 11:31:18 +0800
Subject: [PATCH] input: touchscreen: add the ABS_X and ABS_Y report for gt1x

As the linux platform used the udev/mdev userspace mechanisms,
It needed the ABS_X and ABS_Y when the input event as the touchscreen.
Otherswise the userspace will identify as the keyboard.

Change-Id: I892ed37252d49c5457fe1f25fdd14dabce6ff9cf
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 drivers/input/touchscreen/gt1x/gt1x.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/touchscreen/gt1x/gt1x.c b/drivers/input/touchscreen/gt1x/gt1x.c
index ecb2b1c29438..d701bd5b7193 100644
--- a/drivers/input/touchscreen/gt1x/gt1x.c
+++ b/drivers/input/touchscreen/gt1x/gt1x.c
@@ -499,6 +499,9 @@ static s8 gt1x_request_input_dev(void)
 	input_set_abs_params(input_dev, ABS_MT_TOUCH_MAJOR, 0, 255, 0, 0);
 	input_set_abs_params(input_dev, ABS_MT_TRACKING_ID, 0, 255, 0, 0);
 
+	input_set_abs_params(input_dev, ABS_X, 0, 255, 0, 0);
+	input_set_abs_params(input_dev, ABS_Y, 0, 255, 0, 0);
+
 	input_dev->name = gt1x_ts_name;
 	input_dev->phys = input_dev_phys;
 	input_dev->id.bustype = BUS_I2C;
-- 
2.35.3

